src/VisualStudio/Core/Def/Implementation/TableDataSource/Suppression/VisualStudioDiagnosticListTableCommandHandler.cs

```diff
@@ -176,34 +180,47 @@ private void SetSeverityHandler(object sender, EventArgs args)
             if (pathToAnalyzerConfigDoc != null)
             {
-                var result = _uiThreadOperationExecutor.Execute(
+                var token = _listener.BeginAsyncOperation(nameof(SetSeverityHandler));
+                SetSeverityHandlerAsync(reportDiagnostic, selectedDiagnostic, project).CompletesAsyncOperation(token);
+            }
+        }
+
+        private async Task SetSeverityHandlerAsync(ReportDiagnostic? reportDiagnostic, DiagnosticData selectedDiagnostic, Project project)
+        {
+            try
+            {
+                using var context = _uiThreadOperationExecutor.BeginExecute(
                     title: ServicesVSResources.Updating_severity,
                     defaultDescription: ServicesVSResources.Updating_severity,
                     allowCancellation: true,
-                    showProgress: true,
-                    action: context =>
-                    {
-                        var newSolution = ConfigureSeverityAsync(context.UserCancellationToken).WaitAndGetResult(context.UserCancellationToken);
-                        var operations = ImmutableArray.Create<CodeActionOperation>(new ApplyChangesOperation(newSolution));
-                        var scope = context.AddScope(allowCancellation: true, ServicesVSResources.Updating_severity);
-                        _editHandlerService.Apply(
-                            _workspace,
-                            fromDocument: null,
-                            operations: operations,
-                            title: ServicesVSResources.Updating_severity,
-                            progressTracker: new UIThreadOperationContextProgressTracker(scope),
-                            cancellationToken: context.UserCancellationToken);
-                    });
+                    showProgress: true);
+
+                var newSolution = await ConfigureSeverityAsync(context.UserCancellationToken).ConfigureAwait(false);
+                var operations = ImmutableArray.Create<CodeActionOperation>(new ApplyChangesOperation(newSolution));
+                using var scope = context.AddScope(allowCancellation: true, ServicesVSResources.Updating_severity);
+                await _editHandlerService.ApplyAsync(
+                    _workspace,
+                    fromDocument: null,
+                    operations: operations,
+                    title: ServicesVSResources.Updating_severity,
+                    progressTracker: new UIThreadOperationContextProgressTracker(scope),
+                    cancellationToken: context.UserCancellationToken).ConfigureAwait(false);
 
-                if (result == UIThreadOperationStatus.Completed && selectedDiagnostic.DocumentId != null)
+                if (selectedDiagnostic.DocumentId != null)
                 {
                     // Kick off diagnostic re-analysis for affected document so that the configured diagnostic gets refreshed.
```

Comment:
i don't love that ther'es no async tracking token here.  but that was busted before, so i'm not changing that.

Reply:
i think if we had any test (not sure if we did) it was only checking the sync work we were doing, and not that we kicked off the bg task to 'reanalyze'.  that stays the same here.